include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -487,6 +542,15 @@ MGConstrainedDoFs::get_level_constraint_matrix(const unsigned int level) const
 
 
 
+inline const AffineConstraints<double> &
+MGConstrainedDoFs::get_user_constraint_matrix(const unsigned int level) const
+{
+  AssertIndexRange(level, user_constraints.size());
+  return user_constraints[level];
```

Comment:
also assert that ``user_constraints[level].size()>0``

Reply:
I think we don't want to assert here since we use an empty constraint matrix in the transfer to decide not to reinitialize vectors.

Reply:
oh, sure. That makes sense.